### PR TITLE
Fix subtle pl-order-blocks HTML mustache bug

### LINKED ISSUE #4921: Add vertical lines to pl-order-blocks to show indentation points
(e.g., https://github.com/ace-lab/berk-parsons/blob/79cd59dca70e0ac8156c9386bc8c0b1fcbd94fc0/app/static/vendors/parsons/parsons.js#L1470-L1477, https://github.com/RunestoneInteractive/RunestoneComponents/blob/05d5a5d0a03369fb7726474d68f07b7dbd2d7c26/runestone/parsons/css/parsons.css#L100-L109)

originall suggested by @nathanielweinman over at https://github.com/PrairieLearn/PrairieLearn/issues/3476

### PR DESCRIPTION
**Pull request by Frozemint**
While working on #4921, it became apparent that the HTML mustache layout for `pl-order-blocks` has a subtle bug. Currently, the `order-block-dropzone` div height exceeds the bounds of the parent container div. This is caused by the `ul` within the dropzone being set to height `100%` (IIRC, this was done originally to ensure proper drag and drop functionality: the user can drag anywhere into the dropzone), without accounting for the header div within the container div and thus the `ul` is oversized. 

This wasn't a major issue until I tried to add the indent lines in #4921 using `background`, where the indent lines would visually overflow beyond the container like so, because _the dropzone div itself is_ overflowing beyond its container:

This PR fixes this problem by wrapping the `ul` dropzone in a div, which is sized correctly to have a height to fill up the remainder height of the container div using `display: flex`. As shown, the dropzone is now sized correctly.
image

More work will follow this PR to implement indent lines, as this fix is needed for the indent line implementation to work. 

Edit 1: following the comments by @\SethPoulsen (see https://github.com/PrairieLearn/PrairieLearn/pull/7971#pullrequestreview-1481674366), this PR will now include the implementation for the indent line feature (since the HTML bug fix is a quick small patch). 

Edit 2: The indent line implementation has just been pushed. This PR will close #4921.

### CODE REVIEW

**Review by SethPoulsen — CHANGES_REQUESTED** [review]
Nice catch! 

I must admit I have never noticed this despite having stared at this element for many hours the last few years. 

However, this fix is far too complicated, and I don't like adding extra HTML elements we don't need. 

I just played around with it a little bit myself--removing the line `  height: 100%;` makes it look the way you want, but it makes it harder to drop into the dropzone because you can only drag into the top of it, so that's not ideal.

The ideal thing would be to just make the dropzone automatically fill the remaining height of its parent. From some quick web searching, it looks like this should be possible in 2 or 3 lines of CSS with some stuff related to `flex`. 

TLDR; this should be a 2-3 line CSS only fix, with no changes to HTML.

**Comment by SethPoulsen** [comment]
see https://www.w3docs.com/snippets/html/how-to-make-a-div-fill-the-height-of-the-remaining-space.html

**Comment by Frozemint** [comment]
Yeah, it looks like most of the initial changes was unnecessary. I swear initially the css rules didn't work for some reason until I added the (ultimately unnecessary) HTML. I've went ahead and removed the unnecessary HTML and CSS modifications. I'll add the indent line changes shortly.

**Comment by Frozemint** [comment]
Just pushed the indent grid line changes. Visually they look like this. The grid lines are only visible while dragging a block. The implementation is in javascript, as the number of grid lines depends on the variable `maxIndent`. The JS is also not as elegant as I would like it to be, any help on making the code more elegant would be greatly appreciated.

image

**Comment by SethPoulsen** [comment]
@Frozemint Can you close this PR and open it as a new one? This PR was originally opened just as a bug fix and so the comment history of how it changed it quite confusing.

**Comment by Frozemint** [comment]
Yep, will do.

**Comment by Frozemint** [comment]
Just opened a new pull request #7985 for the changes. To my future self and maintainers looking at this in the future, please read this PR for context, and look at #7985 afterwards.
